### PR TITLE
start adding docker readme for auto-setup

### PR DESCRIPTION
**Pull request by axfelix**
@robholland how's this looking for a start? can push to dockerhub when complete

### CODE REVIEW

File: docker-readme/auto-setup.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+Temporal requires an RDBMS backend; this Docker setup script supports Cassandra (default), MySQL, or PostgreSQL.
+
+Optionally also supports Elasticsearch for [visibility](https://docs.temporal.io/self-hosted-guide/visibility).
+
+## Environment Variables
+
+`DB` (default: `cassandra`)
+
+This variable specifies the type of database you're connecting to. Allowed values are `cassandra`, `mysql`, and `postgres`.
```
Comment (robholland) [inline docker-readme/auto-setup.md:9]: Valid values are: `cassandra`, `postgres12` and `mysql8`.

File: docker-readme/auto-setup.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+`CASSANDRA_USER` (defaults to unset)
+
+This variable specifies your Cassandra username.
+
+`CASSANDRA_PASSWORD` (defaults to unset)
+
+This variable specifies your Cassandra password.
+
+`CASSANDRA_TLS_ENABLED` (defaults to unset)
```
Comment (robholland) [inline docker-readme/auto-setup.md:41]: would probably say "false" here for clarity although you are technically correct.

File: docker-readme/auto-setup.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+`DBNAME` (defaults to `temporal`)
+
+`VISIBILITY_DBNAME` (defaults to `temporal_visibility`)
+
+`DB_PORT` (defautls to `3306`)
+
+This variable specifies the port to connect to MySQL/PostgreSQL on.
+
+`MYSQL_SEEDS` (defaults to unset)
```
Comment (robholland) [inline docker-readme/auto-setup.md:61]: This is the MySQL host name.

File: docker-readme/auto-setup.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+This variable specifies your MySQL username.
+
+`MYSQL_PWD` (defaults to unset)
+
+This variable specifies your MySQL password.
+
+`MYSQL_TX_ISOLATION_COMPAT` (defaults to `false`)
+
+`POSTGRES_SEEDS` (defaults to unset)
```
Comment (robholland) [inline docker-readme/auto-setup.md:73]: Postgresql host name.

File: docker-readme/auto-setup.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+`POSTGRES_TLS_SERVER_NAME` (defaults to unset)
+
+### Elasticsearch
+
+`ENABLE_ES` (defaults to `false`)
+
+`ES_SCHEME` (defaults to `http`)
+
+`ES_SEEDS` (defaults to unset)
```
Comment (robholland) [inline docker-readme/auto-setup.md:101]: Comma-separated list of Elasticsearch nodes.

File: docker-readme/auto-setup.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+`ES_VIS_INDEX` (defaults to `temporal_visibility_v1_dev`)
+
+This variable specifies the name of your Elasticsearch index.
+
+`ES_SEC_VIS_INDEX` (defaults to unset)
+
+This variable specifies the name of your [secondary visibility](https://docs.temporal.io/self-hosted-guide/visibility#dual-visibility) Elasticsearch index.
+
+`ES_SCHEMA_SETUP_TIMEOUT_IN_SECONDS` (defaults to `0`)
```
Comment (robholland) [inline docker-readme/auto-setup.md:127]: 0 meaning no timeout.

File: docker-readme/auto-setup.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+### Server setup
+
+`TEMPORAL_ADDRESS` (defaults to unset)
+
+`TEMPORAL_CLI_ADDRESS` (defaults to unset)
```
Comment (robholland) [inline docker-readme/auto-setup.md:133]: Let's just not mention this variable.

**Review by robholland — CHANGES_REQUESTED** [review]
Looking like a good place to start, a few comments.

File: docker-readme/auto-setup.md [hunk 8]
```diff
@@ -90,22 +104,38 @@ This variable specifies your PostgreSQL password.
 
 `POSTGRES_TLS_ENABLED` (defaults to `false`)
 
+This variable specifies whether you are using TLS to connect to Postgres.
+
 `POSTGRES_TLS_DISABLE_HOST_VERIFICATION` (defaults to `false`)
 
+This variable specifies whether Postgres should skip host key verification (e.g., in case your hostnames have changed recently and you need to work around a false positive red flag).
```
Comment (robholland) [inline docker-readme/auto-setup.md:111]: This is more generally used for server certificates which you can't easily verify, such as Amazon RDS.

**Review by robholland — APPROVED** [review]

**Comment by axfelix** [comment]
@robholland looks like this still needs a review from @temporalio/server and then I'll push it up to dockerhub

**Pull request merged**